python/lsst/summit/utils/efdUtils.py

```diff
@@ -0,0 +1,566 @@
+
+    df = pd.DataFrame()
+    beginTime = timeToLookBefore
+    while df.empty:
+        df = getEfdData(client, topic, begin=beginTime, timespan=-TIME_CHUNKING, noWarn=True)
+        beginTime -= TIME_CHUNKING
+
+    lastRow = df.iloc[-1]
+    commandTime = efdTimestampToAstropy(lastRow['private_sndStamp'])
+
+    commandAge = timeToLookBefore - commandTime
+    if commandAge > staleAge:
+        log = logging.getLogger(__name__)
+        log.warning(f"Component {topic} was last set {commandAge.sec/60:.1} minutes"
+                    " before the requested time")
+
+    return lastRow
+
+
+def getStateAtTime(client, topic, time, warnStaleAfterNMinutes=12*60):
+    """Get the state of a component at a given time.
+
+    Parameters
+    ----------
+    client : `lsst_efd_client.efd_helper.EfdClient`
+        The EFD client to use.
+    topic : `str`
+        The topic to query.
+    time : `astropy.Time`
+        The time at which to get the state.
+    warnStaleAfterNMinutes : `float`, optional
+        The number of minutes after which to consider the data stale and issue
+        a warning.
+
+    Returns
+    -------
+    state : `Enum`
+        The appropriate enumClass for the topic, holding the value it was set
+        to at ``commandTime``.
+    commandTime : `astropy.Time`
+        The time the component was set to the returned state.
+
+    Raises
+    ------
+    ValueError: If the topic is not in the EFD schema.
+    """
+    row = getMostRecentRowWithDataBefore(client=client,
+                                         topic=topic,
+                                         timeToLookBefore=time,
+                                         warnStaleAfterNMinutes=warnStaleAfterNMinutes)
+
+    commandTime = efdTimestampToAstropy(row['private_sndStamp'])
+    # TODO: need to:
+    # a) know which column to get the value from depending on topic, and
+    # b) cast it to the appropriate enum class, and
+    # c) be able to import all the enums, rather than having a few of them them
+    # copied & pasted
+
+    # example: for lsst.sal.MTMount.logevent_azimuthMotionState
+    value = row['state']
+    # state = AxisMotionState(value)
+
+    return value, commandTime
+
+
+def makeEfdClient():
+    """Automatically create an EFD client based on the site.
+
+    Returns
+    -------
+    efdClient : `lsst_efd_client.efd_helper.EfdClient`
+        The EFD client to use for the current site.
+    """
+    if not HAS_EFD_CLIENT:
+        raise RuntimeError("Could not create EFD client because importing lsst_efd_client failed.")
+
+    try:
+        site = getSite()
+    except ValueError as e:
+        raise RuntimeError("Could not create EFD client as the site could not be determined") from e
+
+    if site == 'summit':
+        return EfdClient('summit_efd')
+    if site == 'base':
+        return EfdClient('summit_efd_copy')
+    if site in ['staff-rsp', 'rubin-devl']:
+        return EfdClient('usdf_efd')
```

Comment:
Interesting pattern, that's quite neat. I'd have to call `getSite()` again in the `raise` section though, in order to have the same error message, and it'll be slightly less obvious what's fallen through, and a little less obvious how the function works at first glance, so I'm a little on the fence here. I think there's an argument for a `match` block here instead of how I'm doing it too, but also, I think all this is quite likely to disappear quite soon anyway (if/when the comment you made above about turning it into a class happens), so I'm just going to leave it as-is as I don't think there's anything too wrong with it, and hopefully I'll just remove it in the future.